packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -462,6 +462,23 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     return result;
   }
 
+  public async mineEmptyBlocks(
+    blockCount: BN = new BN(1),
+    intervalInSeconds: BN = new BN(1)
+  ) {
+    // first mine any pending transactions:
+    while (this._txPool.hasPendingTransactions()) {
+      await this.mineBlock();
+    }
```

Comment:
I think this while should also has to take `blockCount` into account. Maybe something like:

```suggestion
    let mined = 0;
    while (blockCount.gtn(mined) && this._txPool.hasPendingTransactions()) {
      await this.mineBlock();
      mined += 1;
    }
```

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/2032/commits/4c6ad94fd0b602827f10565a120bbb58af4117e4